src/main.rs

```diff
@@ -85,6 +87,41 @@ fn run_script(script_file: &String) {
     }
 }
 
+fn is_none(objref: PyObjectRef) -> bool {
+    match objref.borrow().kind {
+        PyObjectKind::None => true,
+        _ => false
+    }
+}
+
+fn shell_exec(vm: &mut VirtualMachine, source: &String, scope: PyObjectRef) {
+    match parser::parse_program(source) {
+        Ok(program) => {
+            for statement in program.statements {
+                match compile::compile_ast(vm, statement) {
```

Comment:
So compile_ast is basically the same as compile with the compilation mode set to 'eval'. See also the builtin function compile in CPython (https://docs.python.org/3/library/functions.html?highlight=compile#compile).

Btw, another hint I got, is that the whole REPL thing can be implemented in python. This is how its done in cpython and pypy. I also believe this is the way to go, since we already have eval and compile builtin functions. This is all you need to implement the repl in python code itself.

Reply:
I think it's probably better to keep it in rust until the python is a bit more complete. We need the ast module, exceptions, io. I suspect it'll end up in python later though.